### ISSUE TITLE
Detect "Use newtype instead of data" for data family instances

### ISSUE THREAD

**Issue by googleson78** [ndmitchell/hlint#1142]
e.g.
```haskell
data family F a

data instance F Int = FInt Int
```
we should suggest `newtype instance` instead here.

**Comment by ndmitchell** [comment]
Is this safe? They can define new instances in other modules, and they might not be newtype convertible.

**Comment by googleson78** [comment]
`ghc` rejects multiple instances:
<details><summary>Example</summary>
<p>

```haskell
blademaster :: /tmp/df » cat Root.hs
{-# LANGUAGE TypeFamilies #-}

module Root where

data family F a

blademaster :: /tmp/df » cat Left.hs
{-# LANGUAGE TypeFamilies #-}

module Left where

import Root

newtype instance F Int = LeftInt Int

blademaster :: /tmp/df » cat Right.hs
{-# LANGUAGE TypeFamilies #-}

module Right where

import Root

newtype instance F Int = RightInt Int

blademaster :: /tmp/df » cat Leaf.hs
module Leaf where

import Left
import Right

blademaster :: /tmp/df » ghc Leaf.hs
[4 of 4] Compiling Leaf             ( Leaf.hs, Leaf.o )

Leaf.hs:1:1: error:
    Conflicting family instance declarations:
      F Int -- Defined in ‘Left’
      F Int -- Defined in ‘Right’
  |
1 | module Leaf where
  | ^

```

</p>
</details>

What is the situation in which this might be unsafe? As long as you want to use both instance you'd have to import them somewhere along the way, which should trigger this error (?)

**Comment by ndmitchell** [comment]
Those are multiple instances of `Int`. What if I define `data instance F Bool = Complicated Int Int Int`?

**Comment by googleson78** [comment]
I think we're having a misunderstanding :thinking: 
I suggest that for this:
```haskell
data family F a

data instance F Int = FInt Int

data instance F Bool = Complicated Int Int Int
```
`hlint` should suggest it be changed to
```haskell
data family F a

newtype instance F Int = FInt Int

data instance F Bool = Complicated Int Int Int
```

**Comment by googleson78** [comment]
`ghc` also handles it as I would expect:
```haskell
{-# LANGUAGE TypeFamilies #-}

module Root where

data family F a

newtype instance F Int = FInt Bool

data instance F Bool = FBool Bool Bool
```
`ghci`:
```haskell
> coerce (FInt True) :: F Bool

<interactive>:6:1: error:
    • Couldn't match representation of type ‘Bool’
                               with that of ‘F Bool’
        arising from a use of ‘coerce’
    • In the expression: coerce (FInt True) :: F Bool
      In an equation for ‘it’: it = coerce (FInt True) :: F Bool
```
(and if I make `F Bool` also be a `newtype` over `Bool` then I can safely coerce between the two)
